base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -564,14 +563,12 @@ private ACMEAuthorization getAuthorization(String authzID, LoadChallenges clc)
         attr = entry.getAttribute(ATTR_STATUS);
         authz.setStatus(attr.getStringValues().nextElement());
 
-        for (String type : IDENTIFIER_TYPES) {
-            attr = entry.getAttribute(ATTR_IDENTIFIER + ";" + type);
-            if (attr != null) {
-                authz.setIdentifier(
-                    new ACMEIdentifier(type, attr.getStringValues().nextElement()));
-                break;
-            }
-        }
+        attr = entry.getAttribute(ATTR_IDENTIFIER);
+        String identifier = attr.getStringValues().nextElement();
+        String[] parts = StringUtils.split(identifier, ":", 2);
+        String type = parts[0];
+        String value = parts[1];
```

Comment:
Here too we should handle `parts.length < 2`.

Reply:
Same as above.